### DISCUSSION TITLE
Direct access to motion properties force and torque

### DISCUSSION THREAD

**Discussion by sherief** [jrouwe/JoltPhysics#333] [General]
Context is synchronizing / diffing state. There's `MotionProperties::SaveState()`, but it does far too much work for my needs - I know for example that restitution, damping, gravity factor aren't changed in my context, but there's no way to directly read force and torque from motion properties or body via public API so I can just serialize these parts and send them over the network. Are there any issues in the design that might cause this to break, or should I just add GetForce() / GetTorque() members to MotionProperties?

**Comment by jrouwe** [comment]
I don't think there are any issues exposing this (other than being aware that after every PhysicsSystem::Update they get reset, so if you want to send the values you need to read them before that point). I've added the accessors for you in https://github.com/jrouwe/JoltPhysics/commit/d76ab0f823eb18eee4d4371962c6110c1de60d2a.
